src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/service/GroupsService.kt

```diff
@@ -0,0 +1,172 @@
+package uk.gov.justice.digital.hmpps.externalusersapi.service
+
+import org.hibernate.Hibernate
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.externalusersapi.jpa.repository.GroupRepository
+import uk.gov.justice.digital.hmpps.externalusersapi.model.Group
+import uk.gov.justice.digital.hmpps.externalusersapi.security.MaintainUserCheck
+
+@Service
+@Transactional(readOnly = true)
+class GroupsService(
+  private val groupRepository: GroupRepository,
+  private val maintainUserCheck: MaintainUserCheck
+) {
+
+  @Throws(GroupNotFoundException::class)
+  fun getGroupDetail(groupCode: String): Group {
+    val requestedGroup =
+      groupRepository.findByGroupCode(groupCode) ?: throw GroupNotFoundException("get", groupCode, "notfound")
+
+    Hibernate.initialize(requestedGroup.assignableRoles)
+    Hibernate.initialize(requestedGroup.children)
+    maintainUserCheck.ensureMaintainerGroupRelationship(groupCode)
+    return requestedGroup
+  }
+
+    /* @Throws(ChildGroupNotFoundException::class)
+     fun getChildGroupDetail(
+       groupCode: String,
```

Comment:
you have tests that still need to be uncommented

Reply:
Removed unrelated (commented) code